crates/accelerate/src/sabre_swap/sabre_dag.rs

```diff
@@ -39,16 +51,18 @@ impl SabreDAG {
         num_clbits: usize,
         nodes: Vec<(usize, Vec<usize>, HashSet<usize>)>,
         node_blocks: HashMap<usize, Vec<SabreDAG>>,
-    ) -> PyResult<Self> {
+    ) -> Self {
```

Comment:
There's no detection in here of failure, though I guess Rust would panic if fed "qubits" or "clbits" that are out of range.  If we're worried about that, we could (possibly should) add proper error handling, but maybe in a separate PR?

Reply:
See #10724 for a possible change here.

Reply:
Oh, I wasn't necessarily suggesting we should be returning an error here. I was more just surprised that nothing complained about us having a result return without an error anywhere in the function. I was assuming clippy or something would see we only returned `Ok(..)` and would have suggested dropping the result (although maybe `PyResult` isn't tracked like that)

Reply:
If it's a public function, I guess clippy couldn't complain about the return value because it could be that it's a defined part of the interface that you (for example) can use the function object as having a particular type in a different crate.  But also the `!` type to make infallible `Result<_, !>` types is relatively new as well, so it may well be that clippy just doesn't care about unnecessary result wrapping.